server/data/data-file/src/main/java/org/apache/james/domainlist/xml/XMLDomainList.java

```diff
@@ -40,6 +43,11 @@
     private final List<String> domainNames = new ArrayList<String>();
     private boolean isConfigured = false;
 
+    @Inject
+    public XMLDomainList(DNSService dns, EnvDetector envDetector) {
```

Comment:
Leak? Could you explain?

I introduced this because I needed indirection for testing (using a mock).